app-devel/llvm-21/01-runtime/defines

```diff
@@ -0,0 +1,126 @@
+PKGNAME=llvm-runtime-21
+PKGSEC=libs
+PKGDEP="gcc-runtime libedit libffi ncurses zlib"
+BUILDDEP="swig doxygen ocaml findlib llvm"
+BUILDDEP__RETRO="llvm"
+BUILDDEP__ARMV4="${BUILDDEP__RETRO}"
+BUILDDEP__ARMV6HF="${BUILDDEP__RETRO}"
+BUILDDEP__ARMV7HF="${BUILDDEP__RETRO}"
+BUILDDEP__I486="${BUILDDEP__RETRO}"
+BUILDDEP__LOONGSON2F="${BUILDDEP__RETRO}"
+BUILDDEP__POWERPC="${BUILDDEP__RETRO}"
+BUILDDEP__PPC64="${BUILDDEP__RETRO}"
+PKGDES="Low Level Virtual Machine Infrastructure (runtime libraries, version 21)"
+
+AB_FLAGS_O3=1
+AB_FLAGS_O3__RETRO=0
+AB_FLAGS_O3__ARMV4="${AB_FLAGS_O3__RETRO}"
+AB_FLAGS_O3__ARMV6HF="${AB_FLAGS_O3__RETRO}"
+AB_FLAGS_O3__ARMV7HF="${AB_FLAGS_O3__RETRO}"
+AB_FLAGS_O3__I486="${AB_FLAGS_O3__RETRO}"
+AB_FLAGS_O3__LOONGSON2F="${AB_FLAGS_O3__RETRO}"
+AB_FLAGS_O3__M68K="${AB_FLAGS_O3__RETRO}"
+AB_FLAGS_O3__POWERPC="${AB_FLAGS_O3__RETRO}"
+AB_FLAGS_O3__PPC64="${AB_FLAGS_O3__RETRO}"
+
+USECLANG=1
+USECLANG__ARMV4=0
+USECLANG__ARMV6HF=0
+USECLANG__LOONGSON2F=0
+# FIXME: libLLVM-16.so linkage failure.
+# relocation R_MIPS_TLS_TPREL_HI16 against `a local symbol' cannot be used
+# when making a shared object; recompile with -fPIC.
+USECLANG__LOONGSON3=0
+
+NOLTO=1
```

Comment:
Shouldn't be needed AFAIK (at least on amd64).  Maybe limit it to problematic platforms.

Reply:
Git history does not provide any clue but https://gcc.gnu.org/PR106943 may be the reason it was needed.

Reply:
Let me give it a go.

Reply:
> Git history does not provide any clue but https://gcc.gnu.org/PR106943 may be the reason it was needed.

I think it was RAM... We only have a couple of machines with enough RAM-per-core to handle this task.